hack/determine_install_upgrade_version.py

```diff
@@ -0,0 +1,55 @@
+import yum
+import sys
+import os.path
+from version_utils import rpm
+
+try:
+	input_pkg = sys.argv[1]
+	input_pkg_obj = rpm.package(input_pkg)
+except IndexError:
+	sys.exit(1)
+except:
+	print "[ERROR] Incorrect package nevra format: " + input_pkg
+	sys.exit(1)
+
+input_pkg_name = str(input_pkg_obj.name)
+input_pkg_version_list = map(int, input_pkg_obj.version.split("."))
+search_version_list = input_pkg_version_list[:]
+search_version_list[1] = search_version_list[1] -1
+search_version = ".".join(map(str, search_version_list)[:2])
+
+yb = yum.YumBase()
+yb.conf.disable_excludes = ["all"]
+
+release_pkgs = []
+pre_release_pkgs = []
+
+available_pkgs = yb.doPackageLists('available', patterns=[input_pkg_name], showdups=True).available
+available_pkgs.sort(key=lambda x: x.version)
+for pkg in available_pkgs:
+    if pkg.version.startswith(search_version):
+    	if pkg.release.startswith("0"):
+    		pre_release_pkgs.append(pkg.vra)
+    	else:
+    		release_pkgs.append(pkg.vra)
+
+def determine_install_upgrade_versions(version_list):
+	write_package_version(version_list[0], "install")
+	if (len(version_list) > 1) and (version_list[0] is not version_list[len(version_list)-1]):
```

Comment:
well I thought that it might happen that there will be pkg duplicity when two repos for some reason would contain the same version of pkg. But not 100% sur eif that could occur...